salt/docker/init.sls

```diff
@@ -24,6 +24,8 @@ include:
         {{ cert | indent(8) }}
     - require_in:
       - docker
+    - onchanges_in:
+      - docker-reload-config
```

Comment:
Where is `docker-reload-config` defined?

Reply:
in the same file at the bottom

Reply:
Ok, found it. So if my previous sugestion doesn't work for you, you could try the equivalent: add the `onchanges` explicitly in `docker-reload-config`:

```SaltStack
docker-reload-config:
  service.running:
    - name: docker
    - reload: True
    - onchanges:
      - file: /etc/docker/daemon.json
 {% for cert_tuple in certs.items() %}
   {% set name, cert = cert_tuple %}
      - file: /etc/docker/certs.d/{{ name }}/ca.crt
 {% endfor %}
```

If this doesn't work, your Salt master is broken ;-)

Reply:
that should work, thanks. will update it